bigquery_etl/cli/query.py

```diff
@@ -685,14 +686,16 @@ def backfill(
         depends_on_past = metadata.scheduling.get("depends_on_past", False)
         # If date_partition_parameter isn't set it's assumed to be submission_date:
         # https://github.com/mozilla/telemetry-airflow/blob/dbc2782fa23a34ae8268e7788f9621089ac71def/utils/gcp.py#L194C48-L194C48
-        date_partition_parameter = metadata.scheduling.get(
-            "date_partition_parameter", "submission_date"
+        scheduling_overrides_dict = json.loads(scheduling_overrides)
+        date_partition_parameter = scheduling_overrides_dict.get(
+            "date_partition_parameter",
+            metadata.scheduling.get("date_partition_parameter", "submission_date"),
         )
```

Comment:
The below nested `get`'s are fairly hard to read at first glance. I'd suggest something like the following:
```suggestion
        scheduling_metadata = dict(metadata.scheduling) 
        scheduling_metadata.update(json.loads(scheduling_overrides))
        date_partition_parameter = scheduling_metadata.get("date_partition_parameter", "submission_date")
```
and so on for `scheduling_parameters`, `date_partition_offset`, etc.

Reply:
I agree.  However, assuming you're doing `dict(metadata.scheduling)` to copy the original scheduling metadata dictionary to avoid modifying the original, I'd recommend using `metadata.scheduling.copy()` instead to make that more explicit.

Reply:
added changes requested...thank you for pointing that out